src/RollOfHonour.Web/Pages/Search/Results.razor

```diff
@@ -224,37 +223,49 @@
         private const int _pageSize = 10;
     public PaginatedList<Memorial> MemorialSearchResults = new();
     public PaginatedList<Person> PersonSearchResults = new();
-    public bool HasResults => (MemorialSearchResults.Any() || PersonSearchResults.Any());
+    public bool HasResults => MemorialSearchResults.Any() || PersonSearchResults.Any();
+    public int ResultsCount => PersonSearchResults.ResultsCount + MemorialSearchResults.ResultsCount;
 
     protected override async Task OnInitializedAsync()
     {
         SetSearchParamsFromQuery();
         PersonSearchResults = new();
         MemorialSearchResults = new();
 
-        if (_selectedQueryType == QueryType.Person)
+        switch (_selectedQueryType)
         {
-            await QueryPeople(false, PageIndex);
-            _regimentFilters = await _searchService.GetRegimentFiltersForSearch(BuildPersonQuery());
-        }
+            case QueryType.Person:
+                await QueryPeople(false, PageIndex);
+                _regimentFilters = await _searchService.GetRegimentFiltersForSearch(BuildPersonQuery());
+                break;
 
-        else if (_selectedQueryType == QueryType.Memorial)
-        {
-            var searchQuery = BuildMemorialQuery();
+            case QueryType.Memorial:
+            {
+                var searchQuery = BuildMemorialQuery();
 
-            Result<PaginatedList<Memorial>> results = await _searchService.MemorialSearch(searchQuery, PageIndex, _pageSize);
+                Result<PaginatedList<Memorial>> results = await _searchService.MemorialSearch(searchQuery, PageIndex, _pageSize);
 
-            if (results.IsSuccess is not true)
-            {
-                return;
-            }
+                if (results.IsSuccess is not true)
+                {
+                    return;
+                }
 
-            if (!results.Value.Any())
-            {
-                return;
+                if (!results.Value.Any())
+                {
+                    return;
+                }
+
+                MemorialSearchResults = results.Value;
+                break;
             }
 
-            MemorialSearchResults = results.Value;
+            case QueryType.Regiment:
+                await QueryPeopleByRegiment(false, PageIndex);
+                _regimentFilters = await _searchService.GetRegimentFiltersForSearchByRegimentName(BuildRegimentPersonQuery());
+                break;
+
+            default:
```

Comment:
Should we redirect to a 404/500 error page? Throwing an exception here doesn't feel very friendly

Reply:
Good spot, arguably it's redundant code, so I've removed it.

Reply:
Is it redundant? What if someone links to a search with a bad URL? What gets returned?

Reply:
In that situation it shows like this, which I thin kit's pretty good.
![image](https://github.com/Nottinghamshire/Roll-of-Honour/assets/783486/552daf36-fb07-4cb3-98cb-fffe46aeb8af)